### PR TITLE
Unit tests of RPC "sign"

### PR DESCRIPTION
**Pull request by ximinez**
Specifically to demonstrate that #229 is resolved.

### CODE REVIEW

**Comment by vinniefalco** [comment]
Suggested commit message:
```
Add more 'sign' tests:

fix #229
```

File: src/test/rpc/JSONRPC_test.cpp [hunk 1]
```diff
@@ -2160,6 +2160,73 @@ class JSONRPC_test : public beast::unit_test::suite
...
 
+        {
+            // 10: Call "sign" with nothing in the open ledger
+            Json::Value toSign;
+            toSign[jss::tx_json] = noop(env.master);
+            toSign[jss::secret] = "masterpassphrase";
+            auto rpcResult = env.rpc("json", "sign", to_string(toSign));
+            auto result = rpcResult[jss::result];
+
+            BEAST_EXPECT(!RPC::contains_error(result));
```
Comment (mellery451) [inline src/test/rpc/JSONRPC_test.cpp]: minor: coding standard says space after `!`, but we have relaxed standards in test, so up to you whether to fix.

File: src/test/rpc/JSONRPC_test.cpp [hunk 2]
```diff
@@ -2160,6 +2160,73 @@ class JSONRPC_test : public beast::unit_test::suite
...
+            toSign[jss::secret] = "masterpassphrase";
+            toSign[jss::fee_mult_max] = 900;
+            auto rpcResult = env.rpc("json", "sign", to_string(toSign));
+            auto result = rpcResult[jss::result];
+
+            BEAST_EXPECT(!RPC::contains_error(result));
+            BEAST_EXPECT(result[jss::tx_json].isMember(jss::Fee) &&
+                result[jss::tx_json][jss::Fee] == "8889");
+            BEAST_EXPECT(result[jss::tx_json].isMember(jss::Sequence) &&
+                result[jss::tx_json][jss::Sequence].isInt());
```
Comment (mellery451) [inline src/test/rpc/JSONRPC_test.cpp]: the expectation is that Sequence is nonzero or > 0 ?

File: src/test/rpc/JSONRPC_test.cpp (hunk 1 above)
Comment (vinniefalco) [inline src/test/rpc/JSONRPC_test.cpp]: That's not one of the relaxations :)

For example, tabs are not okay in test code...

Comment (ximinez) [inline src/test/rpc/JSONRPC_test.cpp]: I'll fix this since I'm making another change.

File: src/test/rpc/JSONRPC_test.cpp (hunk 2 above)
Comment (ximinez) [inline src/test/rpc/JSONRPC_test.cpp]: The parser assigns the `Sequence` as an `intValue` type, and fixing that is beyond the scope of this change. Instead, I'm changing it to `isConvertibleTo(Json::ValueType::uintValue)`.

**Comment by ximinez** [comment]
I just rewrote history to change the commit message, but the first commit is otherwise unchanged.

**Comment by ximinez** [comment]
Squashed

**Comment by seelabs** [comment]
In 0.60.0-b7

**Pull request closed without merging**